src/gt4py/next/program_processors/runners/dace/gtir_to_sdfg_scan.py

```diff
@@ -161,51 +161,32 @@ def _create_scan_field_operator_impl(
     assert field_node_path[-1].dst is field_node
 
-    if inner_map_output_temporary_removed:
-        # The original output of the nested SDFG, the one that would be inside the Map,
-        #  has been deleted and the nested SDFG writes directly into the output.
-        #  In this case we have to adapt the stride of the array inside the nested SDFG.
-        nsdfg_scan = field_node_path[0].src
-        assert isinstance(nsdfg_scan, dace.nodes.NestedSDFG)
-        inner_output_name = field_node_path[0].src_conn
-        inner_output_desc = nsdfg_scan.sdfg.arrays[inner_output_name]
-        assert len(inner_output_desc.shape) == 1
-
-        outside_output_stride = field_desc.strides[scan_dim_index]
-        assert str(outside_output_stride).isdigit()
-        # The stride of the temporary array is constant, so we can just set it on the inside.
-        inner_output_desc.set_shape(
-            new_shape=inner_output_desc.shape, strides=(outside_output_stride,)
-        )
-    else:
-        # The AccessNode on the inside of the Map was not removed but remains there.
-        #  Thus we do not have to update the strides, we do however, make some checks.
-        in_map_temporary_output_field = field_node_path[0].src
-        assert in_map_temporary_output_field == output_edge.result.dc_node
-
-        assert ctx.state.in_degree(in_map_temporary_output_field) == 1
-        assert ctx.state.out_degree(in_map_temporary_output_field) == 1
-        inner_edge = next(iter(ctx.state.in_edges(in_map_temporary_output_field)))
-        nsdfg_scan = inner_edge.src
-        assert isinstance(nsdfg_scan, dace.nodes.NestedSDFG)
-
-        inner_output_name = inner_edge.src_conn
-        inner_output_desc = nsdfg_scan.sdfg.arrays[inner_output_name]
-
-        assert len(inner_output_desc.shape) == 1
-        assert str(inner_output_desc.strides[0]).isdigit()
-        assert inner_output_desc.shape == dataflow_output_desc.shape
-        assert inner_output_desc.strides == dataflow_output_desc.strides
+    # The temporary node which the nested SDFG was writing to has been deleted,
+    # and the nested SDFG will write directly to the result field outside. Thus,
+    # we have to modify the stride of the scan column array inside the nested SDFG
+    # to match the strides outside.
+    nsdfg_scan = field_node_path[0].src
+    assert isinstance(nsdfg_scan, dace.nodes.NestedSDFG)
+    inner_output_name = field_node_path[0].src_conn
+    inner_output_desc = nsdfg_scan.sdfg.arrays[inner_output_name]
+    assert len(inner_output_desc.shape) == 1
+
+    # The result field on the outside is a transient array, so its stride is constant.
+    # We just need to set it on the inside array, without symbol mapping.
+    outside_output_stride = field_desc.strides[scan_dim_index]
+    assert str(outside_output_stride).isdigit()
+    inner_output_desc.set_shape(new_shape=inner_output_desc.shape, strides=(outside_output_stride,))
```

Comment:
I think in that case you could also write:
```suggestion
    inner_output_desc.strides = (outside_output_stride,)
```
It is in my view less confusing.